pandas/core/groupby/groupby.py

```diff
@@ -4290,6 +4290,10 @@ def pre_processor(vals: ArrayLike) -> tuple[np.ndarray, DtypeObj | None]:
                 raise TypeError(
                     "'quantile' cannot be performed against 'object' dtypes!"
                 )
+            elif isinstance(vals.dtype, StringDtype):
+                raise TypeError(
+                    f"{vals.dtype} dtype does not support quantile operations"
```

Comment:
Maybe merge those two errors together, while we are at it? 

That will require updating test `test_groupby_quantile_raises_on_invalid_dtype`, but that actually has an xfail that could probably be removed now (after updating the message to match in the test)

Reply:
good catch on updating that test.  im not sure which two errors you are referring to though.  do you mean sharing the exception message for string+object dtype here?

Reply:
I mean this error and the one a few lines just above for object dtype